appmap/_implementation/detect_enabled.py

```diff
@@ -0,0 +1,182 @@
+        if recording_method:
+            for one_recording_method in RECORDING_METHODS:
+                if one_recording_method == recording_method.lower():
+                    env_var = "_".join(["APPMAP", "RECORD", recording_method.upper()])
+                    if env_var in os.environ:
+                        if os.environ[env_var] == "false":
+                            return [f"{env_var}=false", False]
+                        elif os.environ[env_var] == "true":
+                            return [f"{env_var}=true", True]
+
+        # it's flask
+        message, should_enable = cls.is_flask_and_should_enable()
+        if should_enable == True or should_enable == False:
+            return [message, should_enable]
+
+        # it's django
+        message, should_enable = cls.is_django_and_should_enable()
+        if should_enable == True or should_enable == False:
+            return [message, should_enable]
+
+        if recording_method in RECORDING_METHODS:
+            return ["will record by default", True]
+
+        return ["it's not enabled by any configuration or framework", False]
+
+    @classmethod
+    def is_flask_and_should_enable(cls):
+        if "FLASK_DEBUG" in os.environ:
+            if os.environ["FLASK_DEBUG"] == "1":
+                return [f"FLASK_DEBUG={os.environ['FLASK_DEBUG']}", True]
+            elif os.environ["FLASK_DEBUG"] == "0":
+                return [f"FLASK_DEBUG={os.environ['FLASK_DEBUG']}", False]
+
+        if "FLASK_ENV" in os.environ:
+            if os.environ["FLASK_ENV"] == "development":
+                return [f"FLASK_ENV={os.environ['FLASK_ENV']}", True]
+            elif os.environ["FLASK_ENV"] == "production":
+                return [f"FLASK_ENV={os.environ['FLASK_ENV']}", False]
+
+        return ["it's not Flask", None]
+
+    @classmethod
+    def is_django_and_should_enable(cls):
+        if (
+            "DJANGO_SETTINGS_MODULE" in os.environ
+            and os.environ["DJANGO_SETTINGS_MODULE"] != ""
+        ):
+            try:
+                settings = importlib.import_module(os.environ["DJANGO_SETTINGS_MODULE"])
+            except Exception as exn:
+                settings = None
+                return [
+                    "couldn't load DJANGO_SETTINGS_MODULE={os.environ['DJANGO_SETTINGS_MODULE']}",
+                    False,
+                ]
+
+            if settings:
+                try:
+                    if settings.DEBUG == True:
+                        return [
+                            f"{os.environ['DJANGO_SETTINGS_MODULE']}.DEBUG={settings.DEBUG}",
+                            True,
+                        ]
+                    elif settings.DEBUG == False:
+                        return [
+                            f"{os.environ['DJANGO_SETTINGS_MODULE']}.DEBUG={settings.DEBUG}",
+                            False,
+                        ]
+                except Exception as exn:
```

Comment:
So, then the code should explicitly check for just that. It can do that be either catching *the specific exception* that gets raised in that case, or check to see if `settings` has properties named `DEBUG` or `APPMAP`. Catching `Exception` is too broad, and leads to bugs like the one I originally called out.

Reply:
I see.  Yeah, it should catch a specific exception instead of any exception that comes up.  Fixed.